src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -220,17 +220,21 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
     import ExecutionJsonSupport.ExecutionServiceWorkflowOptionsFormat
     val cromwellSubmission = for {
       (wdl, workflowRecs, wfInputsBatch, wfOpts) <- workflowBatchFuture
-      workflowSubmitResult <- executionServiceDAO.submitWorkflows(wdl, wfInputsBatch, Option(wfOpts.toJson.toString), getUserInfo(credential))
+      workflowSubmitResult <- executionServiceCluster.submitWorkflows(workflowRecs, wdl, wfInputsBatch, Option(wfOpts.toJson.toString), getUserInfo(credential))
     } yield {
-      workflowRecs.zip(workflowSubmitResult)
+      // call to submitWorkflows returns a tuple:
+      val executionServiceKey = workflowSubmitResult._1
+      val executionServiceResults = workflowSubmitResult._2
+
+      (executionServiceKey, workflowRecs.zip(executionServiceResults))
     }
 
-    cromwellSubmission flatMap { results =>
+    cromwellSubmission flatMap { case (executionServiceKey, results) =>
       dataSource.inTransaction { dataAccess =>
         //save successes as submitted workflows and hook up their cromwell ids
         val successUpdates = results collect {
           case (wfRec, Left(success: ExecutionServiceStatus)) =>
-            val updatedWfRec = wfRec.copy(externalId = Option(success.id), status = success.status)
+            val updatedWfRec = wfRec.copy(externalId = Option(success.id), status = success.status, executionServiceKey = Some(executionServiceKey.toString))
```

Comment:
nitpick: it should be Option(executionServiceKey.toString) otherwise you run the risk of getting Some(null)